src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,346 @@
+package edu.harvard.iq.dataverse.util;
+
+/*
+  Eko Indarto, DANS
+  Vic Ding, DANS
+
+  This file prepares the resources used in Signposting
+
+  It requires correspondence configuration to function well.
+  The configuration key used is SignpostingConf.
+  It is a json structure shown below
+
+  useDefaultFileType is an on/off switch during linkset creating time, it controls whether the default type is
+  used, which is always Dataset
+
+  The configuration can be modified during run time by the administrator.
+ */
+
+import edu.harvard.iq.dataverse.*;
+import edu.harvard.iq.dataverse.dataaccess.StorageIO;
+import edu.harvard.iq.dataverse.dataaccess.SwiftAccessIO;
+
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.json.NullSafeJsonBuilder.jsonObjectBuilder;
+
+public class SignpostingResources {
+    private static final Logger logger = Logger.getLogger(SignpostingResources.class.getCanonicalName());
+    SystemConfig systemConfig;
+    DatasetVersion workingDatasetVersion;
+    JsonObject licJsonObj;
+    JsonObject describedByJsonObj;
+    Boolean useDefaultFileType;
+    String defaultFileTypeValue;
+    int maxAuthors;
+    int maxItems;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting) {
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
+            jsonSetting = BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf");
```

Comment:
related to my comment above - would it be simpler to just have a static string for the default in this class? Or to use defaults for the individual entries below (e,g, as useDefaultFileType defaults to true below - actually several fields get defaults that way already).